src/services/UserService.ts

```diff
@@ -112,6 +119,27 @@ export class UserService {
 		return user.toJSONPrivate();
 	}
 
+	public async reportUser(reportingID: string, reportedID: string, options: ReportDataToCreate) {
+		return getConnection().transaction(async entityManager => {
+			if (!reportingID) throw new APIError(HttpCode.NotFound, ReporttUserError.UserNotFound);
+			if (!reportedID) throw new APIError(HttpCode.NotFound, ReporttUserError.UserNotFound);
+			const user = await entityManager.findOneOrFail(User, { id: reportedID })
+				.catch(() => Promise.reject(new APIError(HttpCode.NotFound, ReporttUserError.UserNotFound)));
+
+			const report = new Report();
+			const { currentTime, description } = options;
+			Object.assign(report, { currentTime, description });
+			report.reportedUser = user;
+			report.reportingUser = await entityManager.findOneOrFail(User, { id: reportingID });
+			const reportNew = user.reports ?? [];
+			reportNew.push(report);
+			user.reports = reportNew;
```

Comment:
So since we aren't actually saving the user entity, lines 134-136 shouldn't do anything - could you remove them?